### ISSUE TITLE
Model.objects cannot be defined using QuerySet.as_manager

### ISSUE THREAD

**Issue by orsinium** [typeddjango/django-stubs#1713]
# Bug report

## What's wrong

```python
class SubscriptionPromotion(BasePromotion):
    objects = SubscriptionPromotionQuerySet.as_manager()
```

```
error: Cannot override class variable (previously declared on base class "BasePromotion") with instance variable  [misc]
error: Incompatible types in assignment (expression has type "Manager[BasePromotion]", base class "BasePromotion" defined the type as "ManagerFromPromotionQuerySet[BasePromotion]")  [assignment]
```

## How is that should be

The second error was there before but the first one is newly introduced in 4.2.4. As suggested in https://github.com/typeddjango/django-stubs/issues/1712#issuecomment-1721517677, this code should be explicitly annotated. However, the problem is that if I annotate it like this:

```python
objects: ClassVar[models.Manager[BasePromotion]
```

Then all custom attributes defined in the SubscriptionPromotionQuerySet won't be available in `SubscriptionPromotion.objects`. 

In other words, before django-stubs-ext used to construct on the fly the return type of `QuerySet.as_manager` as a mix of Manager and custom methods but now I can't use it because I have to add explicit annotations, and there is no explicit annotation for this dynamically constructed type.

## System information

- OS: linux
- `python` version: `3.8`
- `django` version: 4.2.4
- `mypy` version: 1.5.1
- `django-stubs` version: 4.2.4
- `django-stubs-ext` version: 4.2.2

**Comment by orsinium** [comment]
We've been using django-stubs for more than a year, and that's the hardest upgrade we had so far, TBH. Automatic inference of `objects` was the main selling point for me to use the `django-stubs-ext` in addition to just `django-stubs`.

**Comment by flaeppe** [comment]
I've noted the problem of inheritance with a base that has managers and such too. I've started working out some stuff on it here: #1699 

However, noting the type you have on your base's manager; `Manager[BasePromotion]`, I would suggest using `._default_manager` attribute where you use your `BasePromotion`, to get around this for now.

Since `._default_manager` aligns perfectly with `Manager[BasePromotion]` (as long as `Manager` here is `django.db.models.manager.Manager`):

https://github.com/typeddjango/django-stubs/blob/dd6076793e526fd55c38eac1fa433facf2a09f92/django-stubs/db/models/base.pyi#L21-L23

Also note that `Manager` has nothing in addition to `BaseManager`. You can verify it by looking at the source:

https://github.com/django/django/blob/814e7bc22062eeae4be9f189e89027e28d5dd290/django/db/models/manager.py#L176-L177

I think this should allow you to avoid having to declare the `objects: ClassVar[...]` for every subclass of your base

**Comment by flaeppe** [comment]
To clarify I meant to remove any `objects: Manager[BasePromotion]` from your `BasePromotion` model and then use `._default_manager` instead of `objects` whenever the type is `type[BasePromotion]`.

**Comment by orsinium** [comment]
Thank you, I'll give it a try on Monday and report back here.

**Comment by orsinium** [comment]
I think now I get what you mean. So, I solved the problem by removing the objects declaration from the base model and copy-pasting it into every submodel. 

Before:

```python
class BasePromotion(models.Model):
    objects = PromotionQuerySet.as_manager()

class Promotion(BasePromotion):
    pass

class SubscriptionPromotion(BasePromotion):
    objects = SubscriptionPromotionQuerySet.as_manager()
```

After:

```python
class BasePromotion(models.Model):
    pass

class Promotion(BasePromotion):
    objects = PromotionQuerySet.as_manager()

class SubscriptionPromotion(BasePromotion):
    objects = SubscriptionPromotionQuerySet.as_manager()
```

It's not perfect, especially if there are lots of submodels, but should do for now.
